File: Backend/Models/Employee.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace Backend.Models;

[Index(nameof(CodeEmployee), IsUnique = true)]
public class Employee
{
    [Key]
    public int IdEmployee { get; set; }

    [Required]
    public long CodeEmployee { get; set; }

    [Required]
    [MaxLength(80)]
    public string NameEmployee { get; set; } = string.Empty;

    [Required]
    [MaxLength(80)]
    public string LastNameEmployee { get; set; } = string.Empty;

    public DateTime? Birthdate { get; set; }

    public char StatusEmployee { get; set; } = 'A';

    public DateTime CreateAt { get; set; } = DateTime.Now;
}

File: Frontend/src/types/employee.ts
import type { PaginatedResponse } from "@typings/api/paginated-response";

export interface Employee {
    idEmployee: number;
    nameEmployee: string;
    lastNameEmployee: string;
    birthdate: string;
    statusEmployee: string;
    createAt: string;
}

export type EmployeeResponse = PaginatedResponse<Employee>;